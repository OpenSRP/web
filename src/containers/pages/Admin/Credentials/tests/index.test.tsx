import React from 'react';
import { shallow, mount } from 'enzyme';
import { createBrowserHistory } from 'history';
import reducerRegistry from '@onaio/redux-reducer-registry';
import { store } from '@onaio/redux-reducer-registry';
import { Provider } from 'react-redux';
import { Router } from 'react-router';
import ConnectedUserCredentials, { UserCredentials } from '..';
import keycloakUsersReducer, {
  fetchKeycloakUsers,
  reducerName,
  KeycloakUser,
} from '../../../../../store/ducks/keycloak';
import { KeycloakService } from '../../../../../services';
import * as fixtures from '../../../../../store/ducks/keycloak/tests/fixtures';

reducerRegistry.register(reducerName, keycloakUsersReducer);

jest.mock('../../../../../configs/env');

const history = createBrowserHistory();

describe('src/containers/pages/Admin/Credentials/', () => {
  const props = {
    keycloakUser: null,
    serviceClass: KeycloakService,
    history,
    location: {
      hash: '',
      pathname: `/user/credentials/${fixtures.keycloakUser.id}`,
      search: '',
      state: undefined,
    },
    match: {
      isExact: true,
      params: {
        userId: fixtures.keycloakUser.id,
      },
      path: '/user/credentials/:userId',
      url: `/user/credentials/${fixtures.keycloakUser.id}`,
    },
  };

  it('does not crash', () => {
    shallow(<UserCredentials {...props} />);
  });

  it('renders correctly', () => {
    store.dispatch(fetchKeycloakUsers(fixtures.keycloakUsersArray as KeycloakUser[]));

    mount(
      <Provider store={store}>
        <Router history={history}>
          <ConnectedUserCredentials {...props} />
        </Router>
      </Provider>
    );
  });
});
