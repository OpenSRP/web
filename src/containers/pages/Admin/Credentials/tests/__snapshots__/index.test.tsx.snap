// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/containers/pages/Admin/Credentials/ renders correctly 1`] = `
Object {
  "children": <ForwardRef(InternalForm)
    initialValues={
      Object {
        "confirm": "",
        "password": "",
        "temporary": false,
      }
    }
    labelCol={
      Object {
        "span": 4,
      }
    }
    onFinish={[Function]}
    wrapperCol={
      Object {
        "span": 16,
      }
    }
  >
    <FormItem
      hasFeedback={true}
      label="Password"
      name="password"
      rules={
        Array [
          Object {
            "message": "Please input your password!",
            "required": true,
          },
        ]
      }
    >
      <Password
        action="click"
        iconRender={[Function]}
        visibilityToggle={true}
      />
    </FormItem>
    <FormItem
      dependencies={
        Array [
          "password",
        ]
      }
      hasFeedback={true}
      label="Confirm Password"
      name="confirm"
      rules={
        Array [
          Object {
            "message": "Please confirm your password!",
            "required": true,
          },
          [Function],
        ]
      }
    >
      <Password
        action="click"
        iconRender={[Function]}
        visibilityToggle={true}
      />
    </FormItem>
    <FormItem
      label="Temporary"
      name="temporary"
      valuePropName="checked"
    >
      <Switch />
    </FormItem>
    <FormItem>
      <Row
        justify="start"
      >
        <Col
          span={4}
        >
          <Button
            block={false}
            className="reset-password"
            ghost={false}
            htmlType="submit"
            loading={false}
          >
            Reset Password
          </Button>
        </Col>
      </Row>
    </FormItem>
  </ForwardRef(InternalForm)>,
  "className": "form-container",
}
`;

exports[`src/containers/pages/Admin/Credentials/ shows validation error if required fields are empty: confirm row class 1`] = `"ant-form-item ant-form-item-has-feedback ant-form-item-has-error"`;

exports[`src/containers/pages/Admin/Credentials/ shows validation error if required fields are empty: confirm row div 1`] = `"ant-row ant-form-item ant-form-item-has-feedback ant-form-item-has-error"`;

exports[`src/containers/pages/Admin/Credentials/ shows validation error if required fields are empty: password div class 1`] = `"ant-row ant-form-item ant-form-item-has-feedback ant-form-item-has-error"`;

exports[`src/containers/pages/Admin/Credentials/ shows validation error if required fields are empty: password row class 1`] = `"ant-form-item ant-form-item-has-feedback ant-form-item-has-error"`;
